test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/PackageSearchCommandTests.cs

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Threading.Tasks;
+using Microsoft.Extensions.CommandLineUtils;
+using Moq;
+using NuGet.CommandLine.XPlat;
+using Xunit;
+using static NuGet.CommandLine.XPlat.PackageSearchCommand;
+
+namespace NuGet.CommandLine.Xplat.Tests
+{
+    public class PackageSearchCommandTests
+    {
+        private CommandLineApplication _app;
+        private Func<ILoggerWithColor> _getLogger;
+        private PackageSearchArgs _capturedArgs;
+        private SetupSettingsAndRunSearchAsyncDelegate _setupSettingsAndRunSearchAsyncDelegate;
+        private string _storedErrorMessage;
+
+        public void SetUp()
+        {
+            _storedErrorMessage = string.Empty;
```

Comment:
I like your idea to use a setup method to keep the tests clean.
You can reduce this further. I'd say there's no need to reassign member fields on each test run. 

This document explains it well:

> xUnit.net creates a new instance of the test class for every test that is run, 

Additionally, recreating the application could be something you could share by using a `CollectionFixture` (also in the doc).

https://xunit.net/docs/shared-context